### ISSUE TITLE
vim-gvim-gtk3 not working since gtk3-3.22.1+69+ga1115c0-1

### ISSUE THREAD

**Issue by Oberon00** [herecura/vim#3]
I file this issue here because gvim worked with gtk3-3.22.1+8+ge11df6c-2 but is broken at least since gk3-3.22.1+69+ga1115c0-1 and no other gtk3 apps seem to be broken. This could also be a genuine Vim bug though. All of the following debugging was done with gtk3-3.22.2+4+gc54f348-1 (the newest version in Arch repos at this time).
EDIT: I use Vim version 8.0.0069-1, but the issue also appears with 8.0.0003-1.

Gvim will show a blank area with some artifacts to the left instead of text:

![gvim](https://cloud.githubusercontent.com/assets/849039/20065741/7d4afc6c-a50f-11e6-8558-1a66aa76af59.png)

In my terminal, the following warning appears 10 times: ``(gvim:107792): Gtk-CRITICAL **: gtk_widget_set_size_request: assertion 'width >= -1' failed`` .  The warning appears again every time I resize the gvim window.

Additionally, if I press `:` I get
```
*** BUG ***
In pixman_region32_init_rect: Invalid rectangle passed
Set a breakpoint on '_pixman_log_error' to debug
```
This seems to be purely a display issue though, because if I (blindly) type `:quit`, gvim closes.

I obtained the following backtrace with `G_DEBUG=fatal-criticals gdb --args gvim -u NONE -f`:

```
(gdb) r
Starting program: /usr/bin/gvim -u NONE -f
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
[New Thread 0x7fffeb24e700 (LWP 117030)]
[New Thread 0x7fffeaa4d700 (LWP 117031)]
[New Thread 0x7fffea24c700 (LWP 117032)]

(gvim:117026): Gtk-CRITICAL **: gtk_widget_set_size_request: assertion 'width >= -1' failed

Thread 1 "gvim" received signal SIGTRAP, Trace/breakpoint trap.
0x00007ffff5f83ff1 in ?? () from /usr/lib/libglib-2.0.so.0
(gdb) bt full

#0  0x00007ffff5f83ff1 in  () at /usr/lib/libglib-2.0.so.0
#1  0x00007ffff5f853fd in g_logv () at /usr/lib/libglib-2.0.so.0
#2  0x00007ffff5f8555f in g_log () at /usr/lib/libglib-2.0.so.0
#3  0x00000000005e29c4 in gtk_form_move_resize (form=0xc1a110, widget=0xc1b0f0, x=0, y=0, w=<optimized out>, h=<optimized out>) at gui_gtk_f.c:1166
#4  0x00000000005d828c in gui_mch_set_text_area_pos (x=<optimized out>, y=<optimized out>, w=<optimized out>, h=<optimized out>) at gui_gtk.c:827
#5  0x00000000005d0139 in gui_position_components (total_width=12) at gui.c:1355
        text_area_x = 0
        text_area_width = -4
        text_area_height = 34
#6  0x00000000005d2ebc in gui_resize_shell (pixel_width=<optimized out>, pixel_height=pixel_height@entry=46) at gui.c:1470
        busy = 1
        pixel_height = 46
        pixel_width = 12
        busy = 1
#7  0x00000000005da78f in form_configure_event (widget=widget@entry=0xc1a110, event=0xe0a030, data=<optimized out>) at gui_gtk_x11.c:4440
        usable_height = 46
#8  0x00007ffff77004ac in _gtk_marshal_BOOLEAN__BOXED (closure=0xdb2fb0, return_value=0x7fffffffd260, n_param_values=<optimized out>, param_values=0x7fffffffd2c0, invocation_hint=<optimized out>, marshal_data=<optimized out>)
    at gtkmarshalers.c:86
        callback = 0x5da740 <form_configure_event>
        cc = 0xdb2fb0
        data1 = 0xc1a110
        data2 = <optimized out>
        v_return = <optimized out>
        __func__ = "_gtk_marshal_BOOLEAN__BOXED"
#9  0x00007ffff6256f75 in g_closure_invoke () at /usr/lib/libgobject-2.0.so.0
#10 0x00007ffff6268f82 in  () at /usr/lib/libgobject-2.0.so.0
#11 0x00007ffff627166f in g_signal_emit_valist () at /usr/lib/libgobject-2.0.so.0
#12 0x00007ffff6271faf in g_signal_emit () at /usr/lib/libgobject-2.0.so.0
#13 0x00007ffff784cc5c in gtk_widget_event_internal (widget=0xc1a110, event=0xe0a030) at gtkwidget.c:7721
        signal_num = <optimized out>
        handled = 0
        event = 0xe0a030
        widget = 0xc1a110
#14 0x00007ffff76ff4d4 in gtk_main_do_event (event=0xe0a030) at gtkmain.c:1830
        event_widget = 0xc1a110
        grab_widget = 0xc1a110
        topmost_widget = <optimized out>
        window_group = 0xdb9d30
        rewritten_event = <optimized out>
        device = 0x0
        tmp_list = <optimized out>
        __func__ = "gtk_main_do_event"
#15 0x00007ffff7211b85 in _gdk_event_emit (event=event@entry=0xe0a030) at gdkevents.c:73
#16 0x00007ffff7242b42 in gdk_event_source_dispatch (source=<optimized out>, callback=<optimized out>, user_data=<optimized out>) at gdkeventsource.c:367
        display = <optimized out>
        event = 0xe0a030
#17 0x00007ffff5f7e587 in g_main_context_dispatch () at /usr/lib/libglib-2.0.so.0
#18 0x00007ffff5f7e7f0 in  () at /usr/lib/libglib-2.0.so.0
#19 0x00007ffff5f7e89c in g_main_context_iteration () at /usr/lib/libglib-2.0.so.0
#20 0x00000000005df2b5 in gui_mch_update () at gui_gtk_x11.c:6471
#21 0x00000000005d3cae in gui_init () at gui.c:720
        wp = 0x0
        recursive = 0
#22 0x00000000005b14d1 in set_termname (term=0x9493d8 "gui") at term.c:1721
        termp = <optimized out>
        builtin_first = <optimized out>
        try = 0
        termcap_cleared = 1
        width = 0
        height = 0
        error_msg = 0x0
        bs_p = <optimized out>
#23 0x00000000005b19bd in termcapinit (name=<optimized out>, name@entry=0x649b08 "builtin_gui") at term.c:2384
        term = <optimized out>
#24 0x00000000005d52f9 in gui_attempt_start () at gui.c:166
        recursive = 1
#25 0x00000000005d53fe in gui_start () at gui.c:110
        old_term = 0x949390 "xterm-256color"
        recursive = 1
#26 0x00000000006111bf in vim_main2 () at main.c:525
#27 0x0000000000438610 in main (argc=<optimized out>, argv=<optimized out>) at main.c:415
        i = <optimized out>
```

**Comment by BlackIkeEagle** [comment]
There is an open PR for vim to fix this https://github.com/vim/vim/pull/1193

But it does not seem to work

**Comment by Oberon00** [comment]
This is fixed as of 8.0.0089 (vim/vim@a859f04b4db651860c07db3587f29906517c552b / herecura/vim@ff451cace38d91e8a598c26ae75c94ef6c2da4db).

**Comment by BlackIkeEagle** [comment]
waw that was fast :)
